src/library/export/coverartcopyworker.cpp

```diff
@@ -0,0 +1,102 @@
+#include "library/export/coverartcopyworker.h"
+
+#include <QBuffer>
+#include <QDebug>
+#include <QFileInfo>
+#include <QMessageBox>
+
+void CoverArtCopyWorker::run() {
+    copyFile(m_coverArtImage, m_coverArtCopyPath);
+
+    if (m_bStop.loadAcquire()) {
+        emit canceled();
+        return;
+    }
+}
+
+void CoverArtCopyWorker::copyFile(
+        const QImage& m_coverArtImage,
+        const QString& m_coverArtCopyPath) {
+    QFileInfo dest_fileinfo(m_coverArtCopyPath);
+
+    if (dest_fileinfo.exists()) {
+        switch (makeOverwriteRequest(m_coverArtCopyPath)) {
+        case OverwriteAnswer::UPDATE:
+        case OverwriteAnswer::CANCEL:
+            stop();
+            return;
+        case OverwriteAnswer::OVERWRITE:
+            break;
+        }
+
+        QFile dest_file(m_coverArtCopyPath);
+        qDebug() << "Removing existing cover art" << m_coverArtCopyPath;
+        if (!dest_file.remove()) {
```

Comment:
This should be done in a more atomic way. 
Copy the new file in the destination folder with a temp name, delete the old file and renale the new file.
We have a class for this.

Reply:
We have https://github.com/mixxxdj/mixxx/blob/2fe0fc58b1be4813b6a3f73288d3468eb775ba6b/src/sources/metadatasourcetaglib.cpp#L635
For this. 
Not sure if this class suits. If yes, you may crate a Par for moving this class into a separate file in the utility folder.

Reply:
Okay 👍 . I will take a look, and if it works, new PR will be opened.

Reply:
I have added this class in the try-out commit.  So I will mark this as a resolved for now.